packages/dds/tree/src/test/feature-libraries/default-schema/schemaChecker.spec.ts

```diff
@@ -0,0 +1,722 @@
+		[Multiplicity.Single, 2, false],
+		[Multiplicity.Sequence, 0, true],
+		[Multiplicity.Sequence, 1, true],
+		[Multiplicity.Sequence, 2, true],
+		[Multiplicity.Optional, 0, true],
+		[Multiplicity.Optional, 1, true],
+		[Multiplicity.Optional, 2, false],
+	];
+	for (const [kind, numberToTest, expectedResult] of multiplicityTestCases) {
+		it(`compliesWithMultiplicity(${numberToTest}, ${Multiplicity[kind]}) === ${expectedResult}`, () => {
+			const actual = compliesWithMultiplicity(numberToTest, kind);
+			assert.equal(actual, expectedResult);
+		});
+	}
+
+	describe("isNodeInSchema", () => {
+		describe("LeafNodeStoredSchema", () => {
+			const numberNode = getValueNode("myNumberNode", 1);
+			const stringNode = getValueNode("myStringNode", "string");
+			const booleanNode = getValueNode("myBooleanNode", false);
+			const nullNode = getValueNode("myNullNode", null);
+			const undefinedNode = getValueNode("myUndefinedNode", undefined);
+			const fluidHandleNode = getValueNode("myFluidHandleNode", new TestFluidHandle());
+			const nodeCases = [
+				numberNode,
+				stringNode,
+				booleanNode,
+				nullNode,
+				fluidHandleNode,
+				undefinedNode,
+			];
+
+			// Making the key of the record a ValueSchema ensures that we'll get compile-time errors if we add new
+			// ValueSchema values but forget to add test cases for them.
+			const testCases = {
+				[ValueSchema.Number]: {
+					schema: new LeafNodeStoredSchema(ValueSchema.Number),
+					positiveNodeType: numberNode,
+				},
+				[ValueSchema.String]: {
+					schema: new LeafNodeStoredSchema(ValueSchema.String),
+					positiveNodeType: stringNode,
+				},
+				[ValueSchema.Boolean]: {
+					schema: new LeafNodeStoredSchema(ValueSchema.Boolean),
+					positiveNodeType: booleanNode,
+				},
+				[ValueSchema.Null]: {
+					schema: new LeafNodeStoredSchema(ValueSchema.Null),
+					positiveNodeType: nullNode,
+				},
+				[ValueSchema.FluidHandle]: {
+					schema: new LeafNodeStoredSchema(ValueSchema.FluidHandle),
+					positiveNodeType: fluidHandleNode,
+				},
+			} satisfies Record<ValueSchema, unknown>;
+
+			for (const [key, testCaseData] of Object.entries(testCases)) {
+				describe(`ValueSchema.${ValueSchema[parseInt(key, 10)]}`, () => {
+					for (const node of nodeCases) {
+						const expectedResult =
+							testCaseData.positiveNodeType === node
+								? SchemaValidationErrors.NoError
+								: SchemaValidationErrors.LeafNode_InvalidValue;
+						const title =
+							expectedResult === SchemaValidationErrors.NoError
+								? "in schema"
+								: "not in schema";
+						it(`${node.type} is ${title}`, () => {
+							const schemaCollection: StoredSchemaCollection = {
+								nodeSchema: new Map([[node.type, testCaseData.schema]]),
+							};
+							assert.equal(
+								isNodeInSchema(node, {
+									schema: schemaCollection,
+									policy: emptySchemaPolicy,
+								}),
+								expectedResult,
+							);
+						});
+					}
+				});
+			}
```

Comment:
This seems like overkill. Its all basically testing that allowsValue works correctly which already has its own unit tests.

It seems more useful to test just a case with matching value, one with out of schema value, and one missing a value (which I don't think you test currently)

Reply:
Simplified